content/en/docs/architecture/ci-operator-steps.md

```diff
@@ -0,0 +1,439 @@
+
+`PipelineImageCacheStep` performs an image build at a predefined place in the
+pipeline based on commands defined in each of the configuration fields mentioned
+above.
+
+#### `RPMServerStep`
+
+- Source: [rpm_server.go](https://github.com/openshift/ci-tools/tree/master/pkg/steps/rpm_server.go)
+- Generated by: `rpm_build_commands`
+- Inputs: the `pipeline:rpms` image
+- Outputs: an RPM web server used by
+  [`RPMImageInjectionStep`](#rpmimageinjectionstep)
+
+`RPMServerStep` creates a `Deployment` and `Route` for a web server that serves
+the RPM packages contained in the `pipeline:rpms` image.
+
+#### `RPMImageInjectionStep`
+
+- Source: [rpm_injection.go](https://github.com/openshift/ci-tools/tree/master/pkg/steps/rpm_injection.go)
+- Generated by: `base_rpm_images`
+- Inputs:
+  - the image in the `pipeline` `ImageStream` created by
+    [`InputImageTagStep`](#inputimagetagstep), named by appending
+    `-without-rpms` to the alias defined in the configuration file
+  - the RPM server created by [`RPMServerStep`](#rpmserverstep)
+- Outputs: an image in the `pipeline` `ImageStream`, named by the alias defined
+  in the configuration file
+
+`RPMImageInjectionStep` builds a new image from the base image tagged by
+[`InputImageTagStep`](#inputimagetagstep). The new image has the public endpoint
+of the route created by [`RPMServerStep`](#rpmserverstep) in its list of RPM
+servers.
+
+#### `ProjectDirectoryImageBuildStep`
+
+- Source: [project_image.go](https://github.com/openshift/ci-tools/tree/master/pkg/steps/project_image.go)
+- Generated by:
+  - `images`
+  - `operator.bundles`
+- Inputs:
+  - the `pipeline:src` image
+  - `images`: the pipeline images referenced in `from` and `inputs`, if defined
+  - `operator.bundles`:
+    - bundle image: the `pipeline:src-bundle` image
+    - index image: the corresponding `ci-index-*-gen` pipeline image
+- Outputs: an image in the `pipeline` `ImageStream`
+
+`ProjectDirectoryImageBuildStep` performs an image build using either an inline
+`Dockerfile` or one from the repository's source code.
+
+#### `OutputImageTagStep`
+
+- Source: [output_image_tag.go](https://github.com/openshift/ci-tools/tree/master/pkg/steps/output_image_tag.go)
+- Generated by: `images`
+- Inputs:
+  - the pipeline image referenced in `from`
+  - the populated `stable` `ImageStream`
+- Outputs: a tag in the `stable` `ImageStream`
+
+`OutputImageTagStep` tags images that were built earlier in the pipeline into
+the stable `ImageStream` so that the test environment reflects the state of the
+release after the changes are integrated.
+
+#### `ImagesReadyStep`
+
+- Source: [images_ready.go](https://github.com/openshift/ci-tools/tree/master/pkg/steps/images_ready.go)
+- Generated by: every configuration
+- Inputs: all steps that import or build images
+- Outputs: the signal that all image imports and builds have finished
+
+`ImagesReadyStep` is a synthetic step added to the graph to generate the correct
```

Comment:
The history is that this was added before `dependencies` and in general solves the problem of "some of the images we build are going to go into the release payload we create, but we don't know which and never cared to find out"